rust/connlib/clients/apple/build-rust.sh

```diff
@@ -6,56 +6,68 @@
 
 set -e
 
+# Sanitize the environment to prevent Xcode's shenanigans from leaking
+# into our highly evolved Rust-based build system.
+for var in $(env | awk -F= '{print $1}'); do
+  if [[ "$var" != "HOME" ]] \
+  && [[ "$var" != "USER" ]] \
+  && [[ "$var" != "LOGNAME" ]] \
+  && [[ "$var" != "TERM" ]] \
+  && [[ "$var" != "PWD" ]] \
+  && [[ "$var" != "SHELL" ]] \
+  && [[ "$var" != "TMPDIR" ]] \
+  && [[ "$var" != "XPC_FLAGS" ]] \
+  && [[ "$var" != "XPC_SERVICE_NAME" ]] \
+  && [[ "$var" != "PLATFORM_NAME" ]] \
+  && [[ "$var" != "CONFIGURATION" ]] \
+  && [[ "$var" != "NATIVE_ARCH" ]] \
+  && [[ "$var" != "ONLY_ACTIVE_ARCH" ]] \
+  && [[ "$var" != "ARCHS" ]] \
+  && [[ "$var" != "SDKROOT" ]] \
+  && [[ "$var" != "OBJROOT" ]] \
+  && [[ "$var" != "SYMROOT" ]] \
+  && [[ "$var" != "SRCROOT" ]] \
+  && [[ "$var" != "TARGETED_DEVICE_FAMILY" ]] \
+  && [[ "$var" != "CONNLIB_TARGET_DIR" ]]; then
+  unset $var
+  fi
+done
+
+# Use pristine path; the PATH from Xcode is polluted with stuff we don't want which can
+# confuse rustc.
+export PATH="/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/homebrew/bin:$HOME/.cargo/bin"
+
 if [[ $1 == "clean" ]]; then
   echo "Skipping build during 'clean'"
   exit 0
 fi
 
-# Default PLATFORM_NAME to macosx if not set.
-: "${PLATFORM_NAME:=macosx}"
-
-export PATH="$HOME/.cargo/bin:$PATH"
-base_dir=$(xcrun --sdk $PLATFORM_NAME --show-sdk-path)
-export INCLUDE_PATH="${base_dir}/usr/include"
-export LIBRARY_PATH="${base_dir}/usr/lib:${LIBRARY_PATH:-}"
-export RUSTFLAGS="-C link-arg=-F$base_dir/System/Library/Frameworks"
-# `-Qunused-arguments` stops clang from failing while building *ring*
-# (but the library search path is still necessary when building the framework!)
-# See https://github.com/briansmith/ring/issues/1332
-export CFLAGS="-L ${LIBRARY_PATH} -I ${INCLUDE_PATH} -Qunused-arguments"
-
-# Borrowed from https://github.com/signalapp/libsignal/commit/02899cac643a14b2ced7c058cc15a836a2165b6d
-# Thanks to @francesca64 for the fix
-if [[ -n "${DEVELOPER_SDK_DIR:-}" ]]; then
-  # Assume we're in Xcode, which means we're probably cross-compiling.
-  # In this case, we need to add an extra library search path for build scripts and proc-macros,
-  # which run on the host instead of the target.
-  # (macOS Big Sur does not have linkable libraries in /usr/lib/.)
-  if [[ "$XCODE_VERSION_MAJOR" -lt "1500" ]]; then
-    # It appears we may not need this workaround with the new linker in Xcode 15, but GitHub Actions
-    # currently has spotty support for Xcode 15 on its macOS 13 VMs, so we'll keep this workaround for now.
-    export LIBRARY_PATH="${DEVELOPER_SDK_DIR}/MacOSX.sdk/usr/lib:${LIBRARY_PATH}"
-  fi
+if [[ -z "$PLATFORM_NAME" ]]; then
+  echo "PLATFORM_NAME is not set"
+  exit 1
 fi
```

Comment:
All of this craziness was never required (except for maybe the `ring` stuff, but that's fixed in 0.17)